src/Components/Components/src/ISerializationModeHandler.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// A service that can infer <see cref="IComponent"/>'s <see cref="PersistedStateSerializationMode"/>.
+/// </summary>
+public interface ISerializationModeHandler
```

Comment:
Why do we need this interface? Can this be done in a different way?

Reply:
The reason for creating this interface is that it was impossible to reference `EndpointHtmlRenderer` in `PersistentComponentState`. Alternative solution was to add some render mode specific absctract methods to the `Renderer` and then override them in `EndpointHtmlRenderer` which didn't feel right.

Reply:
>The reason for creating this interface is that it was impossible to reference EndpointHtmlRenderer in PersistentComponentState.

Have you checked-out Steve's spike? It doesn't require those.

> Alternative solution was to add some render mode specific absctract methods to the Renderer and then override them in EndpointHtmlRenderer which didn't feel right.

That sounds desirable over a separate interface that needs to be implemented in many places.

Reply:
`ISerializationModeHandler` also solves the issue when the code start to run on the circuit or webassembly runtime. When this happens render mode for the components is not available. The reason is that circuit/webassembly runtime uses it own scope and the Renderer's dictionary for component states is not populated. More details this in [this comment](https://github.com/dotnet/aspnetcore/pull/50373#discussion_r1318463261)